magefiles/magefile.go

```diff
@@ -226,3 +237,53 @@ func (CI) isPRPairingRequired() bool {
 
 	return false
 }
+
+func (CI) sendWebhook() error {
+	// TODO we will need to distinguish between webhookTargets for AppStudio QE and HACBS QE
+	const saltSecret = "123456789"
+	const webhookTarget = "https://smee.io/JgVqn2oYFPY1CF"
+
+	var repoURL string
+
+	var repoOwner = os.Getenv("REPO_OWNER")
+	var repoName = os.Getenv("REPO_NAME")
+	var prNumber = os.Getenv("PULL_NUMBER")
+
+	if strings.Contains(jobName, "hacbs-e2e-periodic") {
+		// TODO configure webhook channel for sending HACBS test results
+		klog.Infof("not sending webhook for HACBS periodic job yet")
+		return nil
+	}
+
+	if jobType == "periodic" {
+		repoURL = "https://github.com/redhat-appstudio/infra-deployments"
+		repoOwner = "redhat-appstudio"
+		repoName = "infra-deployments"
+		prNumber = "periodic"
+	} else if repoName == "e2e-tests" || repoName == "infra-deployments" {
+		repoURL = openshiftJobSpec.Refs.RepoLink
+	} else {
+		klog.Infof("sending webhook for jobType %s, jobName %s is not supported", jobType, jobName)
+		return nil
+	}
+
+	path, err := os.Executable()
+	if err != nil {
+		klog.Info(err)
+	}
+
+	hook := &gohooks.GoHook{}
```

Comment:
Maybe a design nitpick it looks to be me that we could make GoHook and it's create/Send an implementation detail of a bigger WebHook interface. Instantiate WebHook with the required data and then tell it send()?

Reply:
I've got an idea about a small improvement of this piece of code. will update this in a next commit

Reply:
let me know what you think: https://github.com/redhat-appstudio/e2e-tests/pull/204/commits/f15e0dbc52d93ea081fb4df3e60acec53e8e06cd

Reply:
Ack! I like it! Looks cleaner to me!